test cases/unit/58 identity cross/stuff.h

```diff
@@ -0,0 +1,27 @@
+#ifdef EXTERNAL_BUILD
+  #ifndef ARG_BUILD
+    #error "External is build but arg_build is not set."
+  #elif defined(ARG_HOST)
+    #error "External is build but arg_host is set."
+  #else
+    #define GOT BUILD
```

Comment:
N.B. `#error` does not cause `get_define` to fail, which surprised me greatly. This made me do the `#elif` chain to get around the lack of early termination.

Reply:
Did you figure out why `get_define()` doesn't fail on `#error`? We should file a bug about that.

Reply:
It looks like because `#error` just messes with exit status and stderrr, and stdout is preprocessed just the same so Meson gets its answer. I guess it's ignoring exit status then.

Reply:
We should fix that (but target 0.51 for it)